[micromega] Deprecate hopefully useless options and flags (#13781)
coq/coq: doc/changelog/04-tactics/13781-deprecate_micromega_options.rst

- **Deprecated:**

>   The micromega option :flag:`Simplex` is deprecated. It is already set by default and will be removed.

Reviewer: ```suggestion
  The micromega option :flag:`Simplex`, which is currently set by default
```

`**Deprecated:**` already covers "is deprecated"
"will be removed" doesn't really add anything, that's what deprecations are about.
